lib/hsa/mcwamp_hsa.cpp

```diff
@@ -763,7 +763,7 @@ class HSAOp : public Kalmar::KalmarAsyncOp {
 
     activity_prof::ActivityProf _activity_prof;
 
-    hsa_status_t _wait_complete_status;
+    std::atomic_flag _dispose_flag;
```

Comment:
I think it would be clearer to replace the type of this flag with `std::once_flag`.  Then, we can replace all the calls to `dispose()` with `std::call_once(_dispose_flag, dispose)`.

Reply:
The overhead of the `std::atomic_flag` is less than the alternative.  I tried to make it clear with the comment at the top of each `dispose()` implementation.